### PR TITLE
Add Unmatch and Block Functionality in Chat,

### LINKED ISSUE #89: Remove Unnecessary Guards

### PR DESCRIPTION
**Pull request by daveking7**
@serainaschraff for UI of chat unmatch functionality
@TwoDigitsOneNumber for Router (removed UserGuard in the progress..)
closes #89

### CODE REVIEW

File: src/components/routing/routers/AppRouter.js [hunk 1]
```diff
@@ -43,19 +40,19 @@ const AppRouter = () => {
 
         {/* user guard */}
         <Route exact path={"/users/:id"}>
-          <UserGuard>
+          <GameGuard>
             <UserPage/>
-          </UserGuard>
+          </GameGuard>
```
Comment (TwoDigitsOneNumber) [inline src/components/routing/routers/AppRouter.js:45]: did you include the game guard here again in order to trigger the check for the token? if so, simple and clean fix! could have probably also used useEffect on history or push url events, but this works too

File: src/components/routing/routers/GameRouter.js [hunk 2]
```diff
@@ -17,36 +18,51 @@ const GameRouter = props => {
 
       {/* menu */}
       <Route exact path={`${props.base}/menu`}>
-        <GameMenu/>
+          <GameGuard>
+              <GameMenu/>
+          </GameGuard>
```
Comment (TwoDigitsOneNumber) [inline src/components/routing/routers/GameRouter.js:23]: did you include the game guard here again in order to trigger the check for the token? if so, simple and clean fix! could have probably also used useEffect on history or push url events, but this works too

File: src/components/ui/UserProfile.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import "styles/views/UserProfile.scss"
 import "styles/ui/Button.scss"
-import {Button} from "../ui/Button";
+import {Button} from "./Button";
```
Comment (TwoDigitsOneNumber) [inline src/components/ui/UserProfile.js:4]: why take the button out of the ui directory?

**Review by TwoDigitsOneNumber — APPROVED** [review]
looks very good, not much else to say :)

File: src/components/ui/UserProfile.js (hunk 3 above)
Comment (daveking7) [reply src/components/ui/UserProfile.js:4]: I didn't take the button out of the ui, but put UserProfile into ui ;)

File: src/components/routing/routers/GameRouter.js (hunk 2 above)
Comment (daveking7) [reply src/components/routing/routers/GameRouter.js:23]: Yes, exactly. I mean, GameGuard does only one thing anyway, so no need to study useEffect on history or push events ;)

**Pull request merged**